deeplake/core/storage/azure.py

```diff
@@ -0,0 +1,273 @@
+        This info is used to fetch new credentials when the temporary 12 hour credentials expire.
+
+        Args:
+            hub_path (str): The deeplake cloud path to the dataset.
+            expiration (str): The time at which the credentials expire.
+            db_engine (bool): Whether Activeloop DB Engine enabled.
+            repository (str, Optional): Backend repository where the dataset is stored.
+        """
+        self.hub_path = hub_path
+        self.tag = hub_path[6:]  # removing the hub:// part from the path
+        self.expiration = expiration
+        self.db_engine = db_engine
+        self.repository = repository
+
+    def subdir(self, path: str, read_only: bool = False):
+        sd = self.__class__(root=posixpath.join(self.root, path))
+        if self.expiration:
+            sd._set_hub_creds_info(
+                self.hub_path, self.expiration, self.db_engine, self.repository
+            )
+        sd.read_only = read_only
+        return sd
+
+    def rename(self, root: str):
+        self.check_readonly()
+        account_name, container_name, root_folder = (
+            root.replace("az://", "").strip("/").split("/", 2)
+        )
+        assert (
+            account_name == self.account_name
+        ), "Cannot rename across storage accounts"
+        assert container_name == self.container_name, "Cannot rename across containers"
+        for blob_name in self._all_keys():
+            source_blob = self.container_client.get_blob_client(blob_name)
+            destination_blob = self.container_client.get_blob_client(
+                f"{root_folder}/{blob_name}"
+            )
+            if destination_blob.exists():
+                raise ValueError(
+                    f"Cannot rename {source_blob.url} to {destination_blob.url} because {destination_blob.url} already exists"
+                )
+            destination_blob.upload_blob_from_url(
+                source_url=source_blob.url, overwrite=False
+            )
+            source_blob.delete_blob()
+        self.root_folder = root_folder
+
+    def get_object_size(self, path: str) -> int:
+        blob_client = self.container_client.get_blob_client(
+            f"{self.root_folder}/{path}"
+        )
+        if not blob_client.exists():
+            raise KeyError(path)
+        return blob_client.get_blob_properties().size
+
+    def get_clients_from_full_path(self, url: str):
+        account_name, container_name, blob_path = self._get_attrs(url)
+        account_url = f"https://{account_name}.blob.core.windows.net"
+        blob_service_client = BlobServiceClient(
+            account_url, credential=self.default_credential
+        )
+        blob_client = blob_service_client.get_blob_client(container_name, blob_path)
+        if not blob_client.exists():
+            raise KeyError(url)
+        return blob_client, blob_service_client
+
+    def get_presigned_url(self, path: str, full: bool = False) -> str:
+        if full:
```

Comment:
please add `_check_update_creds` function to honor credentials update from platform